### DISCUSSION TITLE
CORS and API Issue with Mosapro Custom Integration

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#201] [Problems & Issues] [answered]
My website's url is http://localhost:49985.  I created mosparo project. When I enter http://localhost:50212/index.php, http://localhost:50212/public/index.php/project/ is successfully opened. In general setting page, I input http://localhost:50212 for Connection details-Host and http://localhost:49985/ for General settings-Hosts.
I created Custom Integration page like below attached image:
<img width="965" alt="mosparoformcode" src="https://github.com/mosparo/mosparo/assets/91258926/cf6b7c35-900d-4a57-b55a-137107e371b2">

And Error occurs when I visit this html page.
Error is like attached file.
<img width="1439" alt="mosparo-error" src="https://github.com/mosparo/mosparo/assets/91258926/f258339f-d78f-4707-aaa5-bdc882a09164">

**Comment by zepich** [comment, marked as answer]
Hi @0MAL 

Thank you very much for your question.

Can you remove the trailing slash in the hostname and try again?

![image](https://github.com/mosparo/mosparo/assets/391914/366bbcbe-63a4-48a1-b00f-6df6d972e5f7)

It should work since you've added the `localhost:49985` as host in the general settings in mosparo. But the trailing slash leads to `localhost:50212//api...` which can lead to such issues.

Please let me know if that solved the issue or if you are still having this issue.

Kind regards,

zepich

**Reply by 0MAL** [reply]
Yes, removing trailing slash solves this issue. Thanks.
